### PR TITLE
[ROCm] Enable gluon paged MQA logits on gfx950 (MI355X)

### PR DESCRIPTION
**Pull request by frida-andersson**
### Summary

`rocm_fp8_paged_mqa_logits` in `rocm_aiter_mla_sparse.py` had two separate branches:

- `_ON_GFX942` (MI300X, MI325X): called `deepgemm_fp8_paged_mqa_logits` → gluon single-kernel path - everything else (including MI355X / gfx950): fell through to `deepgemm_fp8_paged_mqa_logits_stage1` → slow two-kernel path (stage1 + `sum(dim=0)`)

The split was introduced in 628c436 (#40871). With Triton >= 3.5.0, `deepgemm_fp8_paged_mqa_logits` inside AITER already dispatches to the gluon kernel for both gfx942 and gfx950 (it asserts on both and sets `cdna_version`/`TotalCuCount` per GPU internally). The `_ON_GFX942`-only guard in vLLM was therefore incorrect — it was accidentally routing MI355X to the slow stage1 path.

**Fix:** 
Extend the condition from `if _ON_GFX942` to `if _ON_GFX942 or _ON_GFX950` so MI355X takes the gluon path.

Before:
Screenshot 2026-05-08 at 13 38 24

After:
Screenshot 2026-05-08 at 13 39 10

### Test plan

- DeepSeek-V3.2 TP€ with HIP graphs and `--block-size 64` on MI355X — confirm `_gluon_deepgemm_fp8_paged_mqa_logits_preshuffle` is dispatched (previously `_stage1`)
- GSM8K 5-shot flexible-extract score

### CODE REVIEW

File: vllm/v1/attention/ops/rocm_aiter_mla_sparse.py [hunk 1]
```diff
@@ -16,9 +16,10 @@
 from vllm.v1.attention.ops.common import pack_seq_triton, unpack_seq_triton
 
 if current_platform.is_rocm():
-    from vllm.platforms.rocm import _ON_GFX942
+    from vllm.platforms.rocm import _ON_GFX942, _ON_GFX950
 else:
     _ON_GFX942 = False
+    _ON_GFX950 = False
```
[the remark this answers is not in the document]
Comment (maeehart) [reply vllm/v1/attention/ops/rocm_aiter_mla_sparse.py:22]: In this case we have to be more specific as Mi308 might have some exceptions.

File: vllm/v1/attention/ops/rocm_aiter_mla_sparse.py [hunk 2]
```diff
@@ -385,7 +386,7 @@ def rocm_fp8_paged_mqa_logits(
         aiter_paged_mqa_logits_module = paged_mqa_logits_module()
 
     if aiter_paged_mqa_logits_module is not None:
-        if _ON_GFX942:
+        if _ON_GFX942 or _ON_GFX950:
```
[the remark this answers is not in the document]
Comment (maeehart) [reply vllm/v1/attention/ops/rocm_aiter_mla_sparse.py:389]: We need both _on_gfx942 and _on_gfx950 elsewhere, so it does not also make sense to apply above recommendation. Also see the comment above.

**Review by maeehart — APPROVED** [review]

**Review by dllehr-amd — APPROVED** [review]
Looks good to me!

**Review by tjtanaa — APPROVED** [review]
LGTM
